src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -187,12 +187,14 @@ private void validateClassRules(List<Throwable> errors) {
      * @return {@code Statement}
      */
     protected Statement classBlock(final RunNotifier notifier) {
-        Statement statement = childrenInvoker(notifier);
-        if (!areAllChildrenIgnored()) {
-            statement = withBeforeClasses(statement);
-            statement = withAfterClasses(statement);
-            statement = withClassRules(statement);
+        if (areAllChildrenIgnored()) {
+            return withIgnores(null, notifier);
         }
+        Statement statement = childrenInvoker(notifier);
+        statement = withBeforeClasses(statement);
+        statement = withIgnores(statement, notifier);
```

Comment:
* Yes, I want the ignored methods to be notified. At the first look, the original implementation on master is confusing, unless you look into implementation of the classes RunBefores and RunAfters. RunBefores execute the given statement *after* the @BeforeClass methods in contrast to RunAfters, which execute the given statement *before* the @AfterClass methods. I will add the suggested test to show that it works.
* No, I am not worried about sending ignore notifications before running the non ignored methods. The order of the running methods, which are not ignored, is the same.